file.go

```diff
@@ -1,37 +1,204 @@
+
+	return myFile, err
 }
 
 func ParseOpenedJsonFile(f *os.File) (file, error) {
-	// TODO: read the given file into your `file` object
-	return file{}, nil
+	var myFile file
+
+	//read complete json file and store in byteValue
+	byteValue, err := ioutil.ReadAll(f)
+
+	if err != nil {
+		fmt.Println(err)
+		return file{}, nil
+	}
+
+	//convert JSON read from file in byteValue to "Data" inside file stucture
+	json.Unmarshal(byteValue, &myFile.data)
+
+	return myFile, err
 }
 
 func ParseCsvFile(path string, hasHeaders bool) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	//open CSV file
+	csvFile, err := os.Open(path)
+
+	if err != nil {
+		return file{}, err
+	}
+
+	defer csvFile.Close()
+
+	//get CSV Reader instance
+	csvReader := csv.NewReader(csvFile)
+	//create an object to hold csv data
+	var CSVData [][]string
+	//read all data into CSVData
+	CSVData, err = csvReader.ReadAll()
+	//if an error occurred
+	if err != nil {
+		return file{}, nil
+	}
+
+	var myFile file
+	myFile.data = make([]map[string]interface{}, len(CSVData))
+
+	//store csv data into file structure
+	i := 0
+	var header []string
+	if hasHeaders {
+		i = 1
+		header = CSVData[0]
+	}
+	for ; i < len(CSVData); i = i + 1 {
+		if hasHeaders {
+			myFile.data[i-1] = make(map[string]interface{})
+			for j := 0; j < len(CSVData[i]); j++ {
+				myFile.data[i-1][header[j]] = CSVData[i][j]
+			}
+		} else {
+			myFile.data[i] = make(map[string]interface{})
+			for j := 0; j < len(CSVData[i]); j++ {
+				myFile.data[i]["key"+fmt.Sprint(j)] = CSVData[i][j]
+			}
+		}
+	}
+	return myFile, err
 }
 
-func ParseCsvJsonFile(f *os.File, hasHeaders bool) (file, error) {
-	// TODO: read the given file into your `file` object
-	return file{}, nil
+func ParseOpenedCsvFile(f *os.File, hasHeaders bool) (file, error) {
+	//creat instance of csv reader
+	csvReader := csv.NewReader(f)
+	var err error
+
+	//create var to hold CSV Data
+	var CSVData [][]string
+	//read the file and store data in CSVData
+	CSVData, err = csvReader.ReadAll()
+	//if an error occurred
+	if err != nil {
+		return file{}, nil
+	}
+
+	var myFile file
+	myFile.data = make([]map[string]interface{}, len(CSVData))
+
+	//store csv data into file structure
+	i := 0
+	var header []string
+	if hasHeaders {
+		i = 1
+		header = CSVData[0]
+	}
+	for ; i < len(CSVData); i = i + 1 {
+		if hasHeaders {
+			myFile.data[i-1] = make(map[string]interface{})
+			for j := 0; j < len(CSVData[i]); j++ {
+				myFile.data[i-1][header[j]] = CSVData[i][j]
+			}
+		} else {
+			myFile.data[i] = make(map[string]interface{})
+			for j := 0; j < len(CSVData[i]); j++ {
+				myFile.data[i]["key"+fmt.Sprint(j)] = CSVData[i][j]
+			}
+		}
+	}
+	return myFile, err
 }
 
 func (f file) Csv(addHeaders bool) []byte {
-	// TODO: return the file data in CSV format
-	return nil
+	if f.data == nil {
+		return nil
+	}
+	var CSVFormat string
+	var header string
+	for key := range f.data {
```

Comment:
You can also use the following syntax:
```suggestion
	for key, value := range f.data {
```